napari/layers/tracks/tracks.py

```diff
@@ -418,59 +419,61 @@ def properties(self) -> Dict[str, np.ndarray]:
         """dict {str: np.ndarray (N,)}: Properties for each track."""
         return self._manager.properties
 
-    @property
-    def properties_to_color_by(self) -> List[str]:
-        """track properties that can be used for coloring etc..."""
-        return list(self.properties.keys())
-
     @properties.setter
     def properties(self, properties: Dict[str, np.ndarray]):
         """set track properties"""
         self.features = properties
 
     @property
-    def graph(self) -> Dict[int, Union[int, List[int]]]:
+    def properties_to_color_by(self) -> List[str]:
+        """track properties that can be used for coloring etc..."""
+        return list(self.properties.keys())
+
+    @property
+    def graph(self) -> Optional[Dict[int, List[int]]]:
```

Comment:
+1 to simplification in typing here and always having a list of int.

Side question, do you believe trying to make that graph immutable would be useful ?